src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/FixedPolicy.cs

```diff
@@ -2,30 +2,47 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Linq;
 
 namespace Azure.Base.Http.Pipeline
 {
-    class FixedPolicy : RetryPolicy {
-        int _maxRetries;
-        TimeSpan _delay;
-        int[] _retriableCodes;
+    internal class FixedPolicy : RetryPolicy
+    {
+        private readonly int _maxRetries;
+
+        private readonly TimeSpan _delay;
+
+        private readonly int[] _retriableCodes;
 
         public FixedPolicy(int[] retriableCodes, int maxRetries, TimeSpan delay)
         {
-            if (retriableCodes == null) throw new ArgumentNullException(nameof(retriableCodes));
+            if (retriableCodes == null)
+            {
+                throw new ArgumentNullException(nameof(retriableCodes));
+            }
 
             _maxRetries = maxRetries;
             _delay = delay;
-            _retriableCodes = retriableCodes;
+
+            _retriableCodes = retriableCodes.ToArray();
             Array.Sort(_retriableCodes);
         }
 
-        protected override bool ShouldRetry(HttpPipelineMessage message, int attempted, out TimeSpan delay)
+        protected override bool ShouldRetry(HttpPipelineMessage message, Exception exception, int attempted, out TimeSpan delay)
         {
             delay = _delay;
-            if (attempted > _maxRetries) return false;
-            if(Array.BinarySearch(_retriableCodes, message.Response.Status) < 0) return false;
-            return true;
+
+            if (attempted > _maxRetries)
+            {
+                return false;
+            }
+
+            if (exception != null)
+            {
+                return true;
```

Comment:
Do we really want to retry on all exceptions? If it's an ArgumentException it's not going to suddenly start working.

Reply:
We can add more logic in the future, I suspect we need a filter delegate or something. This is just the first pass.

Reply:
OK.

Reply:
Apologies for reopening, but given that this is the first pass, may I suggest not retrying as the default behavior for `Exception` and making the filter opt-in?   I'm thinking of scenarios like `OutOfMemoryException`, `ThreadAbortException`, and similar where it could be problematic to retry, rather than just repeating with no hope of success.